gradle/publishing.gradle

```diff
@@ -58,9 +58,41 @@ dependencies {
     mavenDeployer "org.apache.maven.wagon:wagon-ftp:2.8"
 }
 
-afterEvaluate {
-    assert (!projectPomName.isEmpty() && !projectPomDescription.isEmpty())
-    pom {
+uploadArchives { task ->
```

Comment:
@littleaj Thanks for the clarification. I added a commit bed9829 to this PR that makes #592 obsolete. That commit adds a task "generatePom" that generates a pom beside the artifact jar file. This task is a now a dependency of copyLibsToGlobalArtifactsFolder task, so it should cover your way to publish. 
I think it makes sense to keep "uploadArchives" and "install" tasks since those help users of the library to work with local forks and patches. Users would typically want to start using a change internally before the changes have been merged upstream.